src/datasets/summarizations/trend-regression.summarization.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { queryFactory } from './trend-regression.summarization';
+import { TimeSeriesPoint } from '../metas/types';
+
+describe('queryFactory', () => {
+  const overallQuicklyIncreasingPoints: TimeSeriesPoint[] = [];
+  for (let i = 1; i <= 31; i++) {
+    const x = new Date(2020, 6, i);
+    const y = i;
+    overallQuicklyIncreasingPoints.push({ x, y });
+  }
+
+  const weekdayQuicklyIncreasingWeekendConstantPoints: TimeSeriesPoint[] = [];
+  for (let i = 1; i <= 31; i++) {
+    const x = new Date(2020, 6, i);
+    const y = x.getDay() === 0 || x.getDay() === 6 ? 0 : i;
+    weekdayQuicklyIncreasingWeekendConstantPoints.push({ x, y });
+  }
+
+  it('should return all combinations of subsets and linguistic variables.', () => {
+    const nTotalSubsets = 3;
+    const nTrendDynamicOptions = 5;
+    const overallQuicklyIncreasingSummaries = queryFactory(overallQuicklyIncreasingPoints)();
+    expect(overallQuicklyIncreasingSummaries.length).toBe(nTotalSubsets * nTrendDynamicOptions);
+
+    const weekdayQuicklyIncreasingWeekendConstantSummaries = queryFactory(weekdayQuicklyIncreasingWeekendConstantPoints)();
+    expect(weekdayQuicklyIncreasingWeekendConstantSummaries.length).toBe(nTotalSubsets * nTrendDynamicOptions);
+  });
+
+  it('should have validity value between 0 and 1.', () => {
+    const overallQuicklyIncreasingSummaries = queryFactory(overallQuicklyIncreasingPoints)();
+    for (const summary of overallQuicklyIncreasingSummaries) {
+      expect(summary.validity).toBeGreaterThanOrEqual(0.0);
+      expect(summary.validity).toBeLessThanOrEqual(1.0);
+    }
+
+    const weekdayQuicklyIncreasingWeekendConstantSummaries = queryFactory(weekdayQuicklyIncreasingWeekendConstantPoints)();
+    for (const summary of weekdayQuicklyIncreasingWeekendConstantSummaries) {
+      expect(summary.validity).toBeGreaterThanOrEqual(0.0);
+      expect(summary.validity).toBeLessThanOrEqual(1.0);
+    }
+  });
+
+  it('should create summaries describing overall linear trend.', () => {
+    const summaries = queryFactory(overallQuicklyIncreasingPoints)();
+    const isValidSummary = ({ validity }) => validity >= 0.7;
+    const isOverallQuicklyIncreasingSummary = ({ text }) => text.includes('overall') && text.includes('linearly quickly increasing');
+
+    const overallQuicklyIncreasingSummaries = summaries.filter(isOverallQuicklyIncreasingSummary);
+    const otherSummaries = summaries.filter(s => !isOverallQuicklyIncreasingSummary(s));
+
+    expect(overallQuicklyIncreasingSummaries.length).toBe(1);
+    expect(overallQuicklyIncreasingSummaries.every(isValidSummary)).toBeTrue();
+    expect(otherSummaries.some(isValidSummary)).toBeFalse();

```

Comment:
What does this test for? Is there a summary that this dataset should never generate? I think it would be better to be specific, since this might not protect against a breakage and would be hard to debug the problem if it did break.

Here and in the similar expect case in the next test.

Reply:
The summarization is designed to return all summary candidates with the difference in validity values, see the test of `should return all combinations of subsets and linguistic variables.`. So how I verify the returned summaries is by checking the validity of the "expected" summary should be high and others should be low.

Reply:
Do you expect them all to do low? If so can we use otherSummaries.every instead? More specific is better for testing.